packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3082,6 +3083,11 @@ export class ContainerRuntime
 					checkpoint.rollback((message: BatchMessage) =>
 						this.rollback(message.contents, message.localOpMetadata),
 					);
+					// If there are no more pending messages after processing a local message,
+					// the document is no longer dirty.
```

Comment:
reset the dirty state to what it was prev
```suggestion
					// reset the dirty state after rollback, to keep it consistent
```